src/k8s/pkg/k8sd/app/hooks_bootstrap.go

```diff
@@ -270,5 +272,59 @@ func onBootstrapControlPlane(s *state.State, initConfig map[string]string) error
 		return fmt.Errorf("k8s api server did not become ready in time: %w", err)
 	}
 
+	for _, name := range bootstrapConfig.Components {
+		// TODO: This somewhat duplicates the logic in `component.go`
+		switch name {
+		case "network":
+			if err := component.EnableNetworkComponent(s.Context, snap, cfg.Network.PodCIDR); err != nil {
+				return fmt.Errorf("failed to enable network: %w", err)
+			}
+		case "dns":
+			dnsIP, clusterDomain, err := component.EnableDNSComponent(s.Context, snap, "", "", nil)
+			if err != nil {
+				return fmt.Errorf("failed to enable dns: %w", err)
+			}
+			if err := s.Database.Transaction(s.Context, func(ctx context.Context, tx *sql.Tx) error {
```

Comment:
Note: We will need to duplicate/move this piece of code somewhere so we can also set the dns bits from the charm. (Not needed for this PR)

Reply:
Not sure if I understand.
This hook will be executed by microcluster whenever the cluster is bootstrapped. So, it will also be executed if the charm calls the bootstrap endpoint (which the CLI also just calls).

You can also modify the dns bits by using the relevant component endpoint. https://github.com/canonical/k8s-snap/blob/main/src/k8s/pkg/k8sd/api/component.go#L51